tests/collections/asr/test_asr_multitask_model_bpe.py

```diff
@@ -779,3 +780,71 @@ def test_prompted_dataset_canary2(canary2_tokenizer):
         == '<|startofcontext|>s##o##m##ed##e##c##o##d##erc##o##nt##e##x##t<|startoftranscript|><|emo:happy|><|en|><|en|><|pnc|><|noitn|><|timestamp|><|diarize|><|0|>h##el##l##o<|3|><|4|>w##o##r##l##d<|5|><|endoftext|>'
     )
     assert batch.prompted_transcript_lens[i] == 39
+
+
+@pytest.mark.unit
+def test_aed_timestamp_processing():
+    # Create test hypothesis with timestamps
+    hyp = Hypothesis(
+        text="<|10|>hello<|15|> <|20|>world<|25|>",
```

Comment:
Would the model produce ill-formatted timestamps too? e.g. `<|10|>hello<|15|><|16|> <|20|>...`

Probably the current parsing code would crash in such cases, but handling them might be more complex for an uncertain gain...

Reply:
It doesn’t crash as far as I checked, but it ignores words not sandwiched between timestamps since the regex looks for that pattern.